supply.py

```diff
@@ -0,0 +1,50 @@
+import json
+
+from typing import Any, Dict
+from boto3 import client
+
+from constants import base_payload, decimal_places
+
+lambda_client = client('lambda', region_name='us-east-1')
+
+
+def get_status_payload() -> Dict[str, Any]:
+    """ Get payload data from status lambda function
+    """
+    invoke_response = lambda_client.invoke(FunctionName='status-economic')
+    payload = json.loads(invoke_response['Payload'].read().decode('utf-8'))
+    return payload
+
+
+def total_supply(event: Dict[str, Any], context: 'bootstrap.LambdaContext') -> Dict[str, Any]:
+    """ Return the total supply getting data from the status lambda function
+    """
+    status_payload = get_status_payload()
+    if status_payload['statusCode'] != 200:
+        # In case of error we return the same payload
+        return status_payload
+
+    payload = base_payload
+    status_body = json.loads(status_payload['body'])
+    payload['body'] = status_body['total_supply']
+    return payload
+
+
+def circulating_supply(event: Dict[str, Any], context: 'bootstrap.LambdaContext') -> Dict[str, Any]:
+    """ Return the circulating supply getting data from the status lambda function
+    """
+    status_payload = get_status_payload()
+    if status_payload['statusCode'] != 200:
+        # In case of error we return the same payload
+        return status_payload
+
+    status_body = json.loads(status_payload['body'])
+    circulating_supply = status_body['circulating_supply']
+    queryStringParams = event.get('queryStringParameters')
+    if queryStringParams and queryStringParams.get('decimals') == 'true':
+        # If we have decimals: true in the parameter, return as decimal
+        circulating_supply /=  10**decimal_places
+
+    payload = base_payload
+    payload['body'] = circulating_supply
```

Comment:
Where do we set the content type? It should be `text/plain`.

Reply:
This lambda code will be handled by the API Gateway, which expects the message body as a text payload. No need to set the content type. The API Gateway content type will be the default: `application/json`.

Reply:
I agree that it should be `text/plain`.

Reply:
the example API (http://chainz.cryptoid.info/grs/api.dws?q=totalcoins) is `text/plain`, so we should follow that.

Reply:
Ok, changed to `text/plain`